### DISCUSSION TITLE
Limiting Fields Returned in Payload CMS  using REST API

### DISCUSSION THREAD

**Discussion by ab-h-i-n** [payloadcms/payload#9110] [Q&A] [answered]
I'm using Payload CMS and want to fetch only specific fields from a document when making requests through the REST API. For example, instead of retrieving all fields in a user document, I'd like to limit the response to only the name and email fields.

I've tried using the ?fields=name,email query parameter, but it still returns all fields in the response. How can I configure Payload CMS to return only the fields I specify in my REST API requests?

**Comment by andershermansen** [comment]
@ab-h-i-n What you want is the select parameter. It's only available in the Payload CMS 3 beta. You can see the documentation here: https://payloadcms.com/docs/beta/queries/select#rest-api

**Reply by ab-h-i-n** [reply]
Thank you so much for helping me resolve my issue with limiting fields in Payload CMS! Your solution worked perfectly, and it’s made a big difference in streamlining my API responses. I appreciate the time and effort you took to help me out – it was exactly what I needed. Thanks again!

**Reply by ab-h-i-n** [reply]
Thank you for the suggestion! Unfortunately, the ?select parameter isn’t working as expected in my setup. Here’s more information about my environment:

**API Request:**
I'm trying to fetch only the title field using the API:

**http://localhost:3000/api/events?select[title]=true&depth=0**

**Response:**
The response still includes all fields, not just title:

`
{
  "docs": [
    {
      "id": "67309bf46c9bedef78d94234",
      "image": "67309be36c9bedef78d941fb",
      "registrationLink": "gdfgf.vom",
      "title": "dfsd",
      "description": "sdfsd",
      "date": "2034-01-17T11:30:00.000Z",
      "status": "upcoming",
      "location": "sdfsdf",
      "trending": true,
      "createdAt": "2024-11-10T11:41:40.322Z",
      "updatedAt": "2024-11-10T11:41:40.322Z"
    }
  ],
  "totalDocs": 1,
  "limit": 10,
  "totalPages": 1,
  "page": 1,
  "pagingCounter": 1,
  "hasPrevPage": false,
  "hasNextPage": false,
  "prevPage": null,
  "nextPage": null
}`

**Collection Configuration:**
Here’s my events collection configuration:

`
import { isAdmin } from "@/access/isAdmin";
import { CollectionConfig } from "payload";

export const Events: CollectionConfig = {
  slug: 'events',
  access: {
    read: () => true,
    create: isAdmin,
    update: isAdmin,
    delete: isAdmin,
  },
  fields: [
    {
      name: 'image',
      type: 'upload',
      relationTo: 'media',
      required: true,
    },
    {
      name: 'registrationLink',
      type: 'text',
    },
    {
      name: 'title',
      type: 'text',
      required: true,
    },
    {
      name: 'description',
      type: 'textarea',
      required: true,
    },
    {
      name: 'date',
      type: 'date',
      required: true,
    },
    {
      name: 'status',
      type: 'select',
      required: true,
      options: [
        {
          label: 'Upcoming',
          value: 'upcoming',
        },
        {
          label: 'Completed',
          value: 'completed',
        },
      ],
    },
    {
      name: 'location',
      type: 'text',
      required: true,
    },
    {
      name: 'trending',
      type: 'checkbox',
      required: true,
    },
  ],
};
`

**Dependencies:**
Here are the dependencies I’m using in package.json:

`
"dependencies": {
  "@payloadcms/db-mongodb": "beta",
  "@payloadcms/next": "beta",
  "@payloadcms/plugin-cloud": "beta",
  "@payloadcms/richtext-lexical": "beta",
  "@payloadcms/storage-vercel-blob": "3.0.0-beta.109",
  "autoprefixer": "^10.4.20",
  "cross-env": "^7.0.3",
  "graphql": "^16.8.1",
  "next": "15.0.0-canary.160",
  "payload": "beta",
  "postcss": "^8.4.47",
  "react": "19.0.0-rc-5dcb0097-20240918",
  "react-dom": "19.0.0-rc-5dcb0097-20240918",
  "react-server-dom-webpack": "^0.0.1",
  "sharp": "0.32.6"
},
"devDependencies": {
  "@types/node": "^22.5.4",
  "@types/react": "npm:types-react@19.0.0-rc.1",
  "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
  "eslint": "^9.11.1",
  "eslint-config-next": "^14.2.13",
  "tailwindcss": "^3.4.13",
  "typescript": "5.6.2"
},
"engines": {
  "node": "^18.20.2 || >=20.9.0"
}
`
Thank you again for your help! Any further guidance on how to achieve this would be appreciated.

**Reply by andershermansen** [reply, marked as answer]
Your query is correct. I think you are not using new enough beta.
The select was added in [v3.0.0-beta.121](https://github.com/payloadcms/payload/releases/tag/v3.0.0-beta.121)

I suggest that you update to latest. Use https://payload-releases-filter.vercel.app/ to view all breaking changes from your current beta to new beta. While in beta it's best to put the complete version in package.json file for all payload packages to make sure you have latest. The beta tag will not update you automatically.

**Reply by ab-h-i-n** [reply]
Thank you for the suggestion! . I updated to the latest beta version (v3.0.0-beta.127) and the issue with the select function has been resolved ✅.  I'll also check out the [release filter tool](https://payload-releases-filter.vercel.app/) for tracking breaking changes in future updates 🔍.
